### PR TITLE
Tests: Extend TestCatchpointAfterTxns to catch catchpoint write corruption

### PR DESCRIPTION
**Pull request by michaeldiamant**
Extends catchpoint tests to help catch regressions:
* Enriches `testNewLedgerFromCatchpoint` to confirm the writer and reader merkle tries match by comparing merkle trie stats.
* Adds a test case to `TestCatchpointAfterTxns` isolating a scenario that fails _prior to_ https://github.com/algorand/go-algorand/pull/4804.

Notes:
* If folks prefer to _not_ have an isolated test case, I can revert to the initial commit (https://github.com/algorand/go-algorand/commit/a7766298dd6a32a341cea4e1b6d630ce46d2fc0f).
* I ran out of time to convert existing test usage from mocks to `DoubleLedger`.  As is, the PR contains a kludgey if-statement (`Skip invariant check for tests using mocks`).   Even as is, I _think_ the PR presents a value add.

### CODE REVIEW

File: ledger/accountdb.go [hunk 1]
```diff
@@ -132,7 +132,7 @@ var createResourcesTable = []string{
 		PRIMARY KEY (addrid, aidx) ) WITHOUT ROWID`,
 }
 
-var createBoxTable = []string{
+var kvStoreTable = []string{
```
Comment (jannotti) [inline ledger/accountdb.go:135]: Agree with getting rid of "box", but similar variables use naming like `createKvStoreTable`

File: ledger/catchpointwriter_test.go [hunk 2]
```diff
@@ -441,7 +442,7 @@ func TestFullCatchpointWriterOverflowAccounts(t *testing.T) {
 	const maxResourcesPerChunk = 5
 	testWriteCatchpoint(t, ml.trackerDB().Rdb, catchpointDataFilePath, catchpointFilePath, maxResourcesPerChunk)
 
-	l := testNewLedgerFromCatchpoint(t, catchpointFilePath)
+	l := testNewLedgerFromCatchpoint(t, catchpointFilePath, ml.trackerDB().Rdb)
```
Comment (jannotti) [inline ledger/catchpointwriter_test.go:445]: Swap order of last two args, so it matches `testWriteCatchpoint`?

File: ledger/catchpointwriter_test.go [hunk 3]
```diff
@@ -571,6 +572,36 @@ func testNewLedgerFromCatchpoint(t *testing.T, filepath string) *Ledger {
...
+		})
+
+		require.NoError(t, err)
+		return stats
+	}
+
+	// Skip invariant check for tests using mocks that do _not_ update
+	// balancesTrie by checking for zero value stats.
+	ws := balanceTrieStats(catchpointWriterReadAccess)
+	if ws != (merkletrie.Stats{}) {
```
Comment (jannotti) [inline ledger/catchpointwriter_test.go:601]: Why is `ws` allowed to be empty?

File: ledger/catchpointwriter_test.go [hunk 4]
```diff
@@ -653,101 +684,186 @@ func TestExactAccountChunk(t *testing.T) {
...
 func TestCatchpointAfterTxns(t *testing.T) {
 	partitiontest.PartitionTest(t)
 	t.Parallel()
 
-	genBalances, addrs, _ := ledgertesting.NewTestGenesis()
-	dl := NewDoubleLedger(t, genBalances, protocol.ConsensusFuture)
-	defer dl.Close()
+	// Exercises interactions between transaction evaluation and catchpoint
+	// generation to confirm catchpoints include expected transactions.
+	t.Run("chunks", func(t *testing.T) {
```
Comment (jannotti) [inline ledger/catchpointwriter_test.go:697]: Are we gaining something from pushing things into a `t.Run`? It obscures the diff, and I don't think it has the isolation properties you might be hoping for anyway.

File: ledger/catchpointwriter_test.go (hunk 3 above)
Comment (michaeldiamant) [reply ledger/catchpointwriter_test.go:601]: @jannotti See inline comment above `ws` declaration starting with "Skip invariant check".   I also called out in the PR description that the situation feels kludgey, but I exhausted my available time.  I'm open to discussing live.

File: ledger/catchpointwriter_test.go (hunk 4 above)
Comment (michaeldiamant) [reply ledger/catchpointwriter_test.go:697]: @jannotti It's subjective - I think there are 2 discrete test cases here.  I felt they can be left under the same test name (`TestCatchpointAfterTxns`) and separated by `t.Run`.

The PR's initial commit shows everything inline as 1 test case without `t.Run`.  As mentioned in the PR description, I'm willing to remove `t.Run` if reviewers feel strongly.

File: ledger/accountdb.go (hunk 1 above)
Comment (michaeldiamant) [reply ledger/accountdb.go]: @jannotti Sorry - I missed the existing convention.  I think it's fixed in https://github.com/algorand/go-algorand/pull/4818/commits/0ea6c872974eb6425ac910aacb17cd3d70089b4d.

File: ledger/catchpointwriter_test.go (hunk 2 above)
Comment (michaeldiamant) [reply ledger/catchpointwriter_test.go]: @jannotti https://github.com/algorand/go-algorand/pull/4818/commits/32529a07c77263c1b8bb3b596a4b90122ece9089 reorders arguments to bring in the feedback.

File: ledger/catchpointwriter_test.go (hunk 4 above)
Comment (michaeldiamant) [reply ledger/catchpointwriter_test.go]: @jannotti Per our verbal, https://github.com/algorand/go-algorand/pull/4818/commits/c9425ed9d67ae0e8e4d301f7b4e844254914a6b9 removes `t.Run` in favor of discrete tests.

**Review by jannotti — APPROVED** [review]

**Pull request merged**